### PR TITLE
fix: update performance when used with other preset

### LINKED ISSUE #92: Plugin very slow when using with other preset
Hi,

I notice performances issues when using this plugin with other eslint config preset such as `eslint-config-airbnb-base`.

For instance, I got 7 svelte files and it tooks 11seconds to lint with 'airbnb-base' active and only 1s when I removed it.

Here is my .eslintrc file :

```js
module.exports = {
  env: {
    browser: true,
    es2021: true,
  },
  extends: ['airbnb-base', 'prettier'],
  ignorePatterns: ['public/build'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 2019,
    sourceType: 'module',
  },
  plugins: ['svelte3', '@typescript-eslint'],
  overrides: [
    {
      files: ['*.svelte'],
      processor: 'svelte3/svelte3',
      rules: {
        'import/first': ['off'],
        'import/no-duplicates': ['off'],
        'import/no-mutable-exports': ['off'],
      },
    },
    {
      files: ['.eslintrc.js', '.prettierrc.js'],
      rules: {
        'global-require': ['off'],
        'import/no-extraneous-dependencies': ['off'],
      },
    },
  ],
  settings: {
    'svelte3/typescript': require('typescript'),
  },
  rules: {},
};

```

Can you take a look ? Thanks  🙏

### PR DESCRIPTION
**Pull request by mathieumure**
Fix #92 

I figured out that passing typescript module in settings is too heavy and produces overhead in other preset such as `eslint-config-airbnb-base`.

In this PR, I load typescript only once to avoid overhead.

### CODE REVIEW

**Comment by dummdidumm** [comment]
This should be a backwards compatible change to not break existing users, so it would need to check if a boolean was passed in. Also I don't like that now there's a require of TypeScript inside the code base, the API was designed that way to avoid that. I instead suggest something like `'svelte3/typescript': () => require('typescript')` which would also achieve the lazy loading behavior you are looking for.

The other thing is I don't quite understand how this will help performance. If you use TypeScript, it will be used either way at some point, so if you pay the cost upfront or later on shouldn't matter right?

**Comment by Conduitry** [comment]
I skimmed @dummdidumm's comment, and then was about to say "If we have to lazy load, I'd much prefer something like `() => require('typescript')` where it remains the config file's responsibility to do the loading" and then I saw that that is exactly what he suggested.

I too am confused about where the overhead would be coming from here. Can you explain more what's going on? Is the config file re-evaluated (without a `require` cache) a bunch of times for different plugins?

File: package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
 		"eslint": ">=6.0.0",
 		"rollup": "^2",
 		"sourcemap-codec": "1.4.8",
-		"svelte": "^3.2.0",
```
Comment (mathieumure) [inline package.json:43]: 🤔 Master test are broken with svelte "3.35.0". This is a temporary fix just to be sure all the tests working fine.

**Comment by mathieumure** [comment]
Hello @Conduitry and @dummdidumm,

Thanks for the quick reply 👍
You are right and I just realized that my code is not backward compatible after creating the PR 😅
I just correct accordingly to your suggestions.

I'm still trying to figure out what are the rules slowing down by this setting.
My guess is that some rule recursively runs through all keys in settings and the `require('typescript') is really big.
So far, I find that it's linked to import related rules but I don't have the reason why.

File: package.json (hunk 1 above)
Comment (mathieumure) [reply package.json:43]: I open an issue for that #94

File: test/samples/typescript-block-filenames/.eslintrc.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ module.exports = {
 		},
 	],
 	settings: {
-		'svelte3/typescript': require('typescript'),
+		'svelte3/typescript': () => require('typescript'),
```
Comment (dummdidumm) [inline test/samples/typescript-block-filenames/.eslintrc.js:20]: Could you revert the changes to the existing tests and add a new test which tests the lazy loading? Like `typescript-lazy-load`

Comment (mathieumure) [reply test/samples/typescript-block-filenames/.eslintrc.js]: Yes nice idea 👍

Comment (mathieumure) [reply test/samples/typescript-block-filenames/.eslintrc.js]: done 😉

**Comment by dummdidumm** [comment]
In #95 it came up to do the `true` option like you initially proposed. In that issue @Conduitry seemed to be okay now with going that route. If he confirms, then you could switch back the implementation to the first one. Sorry for the back and forth 😅

**Comment by mathieumure** [comment]
@dummdidumm ha ha ha ok. I'll do it when @Conduitry will confirm it 😉.

**Comment by mathieumure** [comment]
@dummdidumm I add a boolean option with the functional one. Tell if you are okay with this.

**Comment by JounQin** [comment]
> @dummdidumm I add a boolean option with the functional one. Tell if you are okay with this.

Or add string option too?

**Comment by mathieumure** [comment]
> > @dummdidumm I add a boolean option with the functional one. Tell if you are okay with this.
> 
> Or add string option too?

What do you intend to do with this string ? 🤔  Do you have any use case  ?

**Comment by JounQin** [comment]
@mathieumure I want to use [`ttypescript`](https://github.com/cevek/ttypescript) instead.

**Comment by mathieumure** [comment]
@JounQin 🤔 I think we can achieve this in another PR because IMO it's a new feature.

**Comment by mathieumure** [comment]
@Conduitry did you have the time to check this PR ?

**Review by dummdidumm — APPROVED** [review]

**Comment by Conduitry** [comment]
This has been released (with some adjustments in) 3.2.0. Thank you!
